rts/motoko-rts/src/stabilization/layout/stable_bigint.rs

```diff
@@ -0,0 +1,85 @@
+use crate::stabilization::layout::{checked_to_usize, write_padding_u64};
+use crate::stabilization::reader_writer::{ScanStream, StableMemorySpace, WriteStream};
+use crate::stabilization::StableMemoryAccess;
+use crate::tommath_bindings::{mp_digit, mp_int};
+use crate::types::{BigInt, Bytes, Obj, Value, TAG_BIGINT};
+
+use super::{checked_to_u32, round_to_u64, Serializer, StableValue, StaticScanner};
+
+// TODO: Use a format that is independent of Tom's math library, e.g. by using LEB128/SLEB128 encoding.
```

Comment:
I did a bit of digging and for bignum serialization, we could also use the functions provided by LibTomMath 

See section 10.2 of the pdf manual https://github.com/libtom/libtommath/releases/download/v1.2.0/bn-1.2.0.pdf

There's to/from_sbin (bigendian) and libgmp style pack unpack available. Might actually be easier/cheaper than sleb encoding, plus there's a function for determining the size in bytes upfront.

Reply:
Thank you. This seems interesting. 
A challenge that I see is that these functions require temporary pre-allocated buffer memory with a size depending on the bignum. The serialization (`to_sbin` calling `to_ubin`) even allocates an additional internal copy of the bignum. This could be problematic for large-sized bignums, but I am not sure how relevant this is in practice.

Reply:
**Update**: I implemented the more portable serialization using the suggested LibTomMath functions. As mentioned, the only disadvantage are temporary extra allocations involved, but maybe, for moderately-sized big numbers this is acceptable.

Reply:
**Another update**: I now implemented a custom direct serialization of a bignum to little endian arbitary-length byte sequence, without using Tom's math library. As a result, no temporary allocations are needed.